app/mod_api/controllers.py

```diff
@@ -215,11 +215,22 @@ def get_resource(publisher, package, resource):
         if path.endswith('csv'):
             resource_key = metadata.build_s3_key(resource + '.csv')
             data = metadata.get_s3_object(resource_key)
-            return Response(data, mimetype='text/csv'), 200
+            def generate():
+                for row in data.splitlines():
```

Comment:
Use of `.splitlines`:

- Will break if a cell has line endings (which will definitely happen)
- Will cause entire contents of file to be read into memory

Consider using https://github.com/frictionlessdata/tabulator-py for robust tabular data handling, including streaming read over HTTP.